add description for treasury contract (#33)
OlympusDAO-Education/Documentation: contracts.md

## Treasury

https://etherscan.io/address/0x886CE997aa9ee4F8c2282E182aB72A705762399D

> The treasury contract is a simple vault implementation holding all RFV funds of

Reviewer: ```suggestion
The treasury contract is a simple vault implementation holding all the funds collected by
```

Reviewer: The reason behind this change is that in addition to DAI, the treasury also holds the full amount of SLP token, not just its RFV.

the protocol. If for instance a user purchases a DAI bond, the bonded DAI is
fully taken in by the treasury in return of the market equivalent of OHM bonded

> for. Additional OHM minted is being sent to the distributor contract.

Reviewer: ```suggestion
for. New OHM will be minted based on the RFV of the treasury assets.
```
